src/sage/groups/perm_gps/permgroup_named.py

```diff
@@ -3435,3 +3438,81 @@ def codegrees(self):
+        contains a database of groups, identified by a GAP id,
+        a pair ``[n,k]`` consisting of ``n``, the order of the
+        group, and ``k``, an index determining the group
+        specifically. This class can construct the group
+        as a permutation group from this data.
+
+        INPUT:
+
+        - ``order`` -- the order of the group
+
+        - ``gap_id`` -- the numerical index in the GAP Id of the group
+
+        Generators may be obtained through the :meth:`gens` method.
+        These could change for a particular group in later releases
+        of GAP. In many instances the degree of the constructed group
+        ``SmallPermutationGroup(n,k)`` will be a permutation group on
+        `n` letters, but this will not always be true.
+
+        EXAMPLES::
+
+            sage: G = SmallPermutationGroup(12,4); G
+            Group of order 12 and GAP Id 4 as a permutation group
+            sage: G.gens()
+            ((1,2)(3,5)(4,10)(6,8)(7,12)(9,11),
+            (1,3)(2,5)(4,7)(6,9)(8,11)(10,12),
+            (1,4,8)(2,6,10)(3,7,11)(5,9,12))
+            sage: G.character_table()
+            [ 1  1  1  1  1  1]
+            [ 1 -1 -1  1  1 -1]
+            [ 1 -1  1  1 -1  1]
+            [ 1  1 -1  1 -1 -1]
+            [ 2  0 -2 -1  0  1]
+            [ 2  0  2 -1  0 -1]
+            sage: all(SmallPermutationGroup(n,k).id()==[n,k] for n in [1..64] for k in [1..oeis(1)[n]])
+            True
+            sage: H = SmallPermutationGroup(6,1)
+            sage: H.is_abelian()
+            False
+            sage: [H.centralizer(g) for g in H.conjugacy_classes_representatives()]
+            [Subgroup generated by [(1,2)(3,6)(4,5), (1,3,5)(2,4,6)] of (Group of order 6 and GAP Id 1 as a permutation group),
+            Subgroup generated by [(1,2)(3,6)(4,5)] of (Group of order 6 and GAP Id 1 as a permutation group),
+            Subgroup generated by [(1,3,5)(2,4,6), (1,5,3)(2,6,4)] of (Group of order 6 and GAP Id 1 as a permutation group)]
+        """
+        self._order = order
+        self._gap_id = gap_id
+        self._gap_small_group = libgap.SmallGroup(order,gap_id)
+        self._gap_permutation_group = self._gap_small_group.IsomorphismPermGroup().Image(self._gap_small_group)
+        PermutationGroup_generic.__init__(self, gap_group=self._gap_permutation_group)
+
+    def _repr_(self):
+        r"""
+        EXAMPLES::
+
+            sage: G = SmallPermutationGroup(12,4); G
+            Group of order 12 and GAP Id 4 as a permutation group
+        """
+        return "Group of order %s and GAP Id %s as a permutation group"%(self._order, self._gap_id)
+
+    def gap_small_group(self):
+        r"""
+        Gap realizes some small groups as PermutationGroup, others as PcGroups
+        (polycyclic groups). The :class:`SmallPermutationGroup` class always
+        returns a PermutationGroup, but in the process of creating this group
+        a gap SmallGroup is generated. This method returns that group.
```

Comment:
```suggestion
        Return the GAP small group object corresponding to ``self``.
        
        Gap realizes some small groups as PermutationGroup, others as PcGroups
        (polycyclic groups). The :class:`SmallPermutationGroup` class always
        returns a PermutationGroup, but in the process of creating this group
        a gap SmallGroup is generated. This method returns that group.
```

Reply:
I made most of the changes requested by Travis, except one comment above that I am asking for clarification on.

Unfortunately if I add a testsuite to __init__ (currently commented out) there are many failures.

Reply:
I meant to put my comment below here.

You also have underindented the `EXAMPES::` block I think.

Reply:
> You also have underindented the EXAMPES:: block I think.

Fixed.

Reply:
The problem comes from the fact the permutation group expects that `_order` is a method. There are 2 ways to fix this. The first would be to change `_order` to, e.g., `_group_order` and override the `_order()` method to simply return `_group_order`. The second would be to override `order()` to simply return `self._order`. The method `_order()` only seems to be used in the `order()` method. So I would go with option 2.

Reply:
So I added this at the end of the class and the `TestSuite` passed:
```python

    def order(self):
        return self._order

    cardinality = order
```

Reply:
> So I added this at the end of the class and the `TestSuite` passed:

I fixed this alternatively by just renaming the `_order` attribute `_n` and the testsuite passes.

>
> ```python
>     def order(self):
>         return self._order
> 
>     cardinality = order
> ```